src/main/java/net/minecraftforge/client/model/pipeline/LightUtil.java

```diff
@@ -194,15 +194,15 @@ else if(type == VertexFormatElement.EnumType.UINT)
                 }
                 else if(type == VertexFormatElement.EnumType.BYTE)
                 {
-                    to[i] = ((float)(byte)bits) / mask * 2;
+                    to[i] = ((float)(byte)bits) / (mask >> 1);
                 }
                 else if(type == VertexFormatElement.EnumType.SHORT)
                 {
-                    to[i] = ((float)(short)bits) / mask * 2;
+                    to[i] = ((float)(short)bits) / (mask >> 1);
                 }
                 else if(type == VertexFormatElement.EnumType.INT)
                 {
-                    to[i] = ((float)(bits & 0xFFFFFFFFL)) / 0xFFFFFFFFL * 2;
+                    to[i] = (float)((double)(bits & 0xFFFFFFFFL) / (0xFFFFFFFFL >> 1));
```

Comment:
How does the `double` precision have an effect here?

Reply:
See http://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.2.
In particular the points about `int` to `float` conversions - large values won't be converted accurately.
So, we cast the `int` to `double`, do the floating-point divisions, and then reduce the result to a `float`.

Reply:
I'm just curious, but does it have a noticeable effect visually or were you just being thorough?

Reply:
Just being thorough, don't think vanilla really uses `INT` anywhere. It also makes the code consistent with the `UINT` case.